### DISCUSSION TITLE
Old `<script>` tags without `type="module"`

### DISCUSSION THREAD

**Discussion by shishkin17** [vikejs/vike#1437] [Discussion] [answered]
After pre-rendering, scripts are loaded with type="module" and rel="modulepreload". Is it possible to configure to load them in an old way not as a module for a more browser compatibility?

**Comment by brillout** [comment, marked as answer]
I'm not sure this should be a priority for the Vike team. What's your use case and what browsers don't support them?

That said, I believe https://github.com/vitejs/vite/tree/main/packages/plugin-legacy does what you want.

**Reply by shishkin17** [reply]
Looks like that what I need, thanks!

**Reply by brillout** [reply]
Let us know how it goes. The latest version may have introduced a regression related to the legacy plugin.

(Also we're looking for [sponsors](https://github.com/vikejs/vike/issues/1350) in case your company is up for it.)

**Reply by shishkin17** [reply]
Just gave it a quick try, it seems it will not work with vike out of the box. It does generate legacy chunks, but because vike creates html files itself they lack these fallback nomodule injections:

![2024-01-16_11-58-18](https://github.com/vikejs/vike/assets/48315811/ac12fffd-b371-456a-bcdb-78f04e750f72)

Anyway, this is not a feature request)
I was curious about the possibility to support some older browsers.
This info can be also helpful for someone else thinking about it.

**Reply by brillout** [reply]
I know some users have been using the legacy pugin with vite-plugin-ssr (previous Vike project name) in production in the past. E.g. Lukas from Rollup used it in his company.

Have you seen the previous discussions about using Vike with the legacy plugin? IIRC there is even a working demo linked somewhere in these discussions. But, as said, Vike's latest versions may have introduced a regression.

A possibility would be to include a legacy plugin test to Vike's CI. But for that I'd ask you to answer following questions.

> I'm not sure this should be a priority for the Vike team. What's your use case and what browsers don't support them?

**Reply by shishkin17** [reply]
Oh, you're right, that topic was discussed before. I guess that is the solution you meant: https://github.com/vikejs/vike/discussions/928. It's really tricky)

I don't have a special case, I wanted to see if it is possible to support older browsers, but now I see it is not relevant in 2024.
